prover/strategies/apply-equation.md

```diff
@@ -1,26 +1,28 @@
+# apply-equation
```

Comment:
It is important that we leave new lines between `# headings` and text. Otherwise line numbers in the markdown and `.k` file stop matching up.

Please add a `TODO` about splitting this up as we discussed.